haskell-open-source.md

```diff
@@ -0,0 +1,35 @@
+## Best Practices
+
+1. If authoring a distinct package of reasonable size and generic utility,
+   prefer starting a separate, public repository utilized as a Git dependency
+
+When working in a public repository:
+
+1. Keep a linear Git history with rich, meaningful commit messages
+
+   Configure GitHub to only allow Squash and Rebase strategies.
```

Comment:
You really are gunning for controversy :smile: 

:+1:

Reply:
heh I love merge commits. It's much easier to see a merge commit like "Merged pull request XXX" so I know where a chunk of code came from vs an uninformative cleanup commit like "fixed linting issues" that inevitably come at the end of a PR :man_shrugging: Feel free to keep this but I don't think its necessary at all for an OSS guide.

Reply:
> vs an uninformative cleanup commit like "fixed linting issues" that inevitably come at the end of a PR

Yeah, that's been an unfortunate risk, but it's why Squash is there: if one doesn't want to take the time to rebase about fixup commits, they can just choose Squash.

It's no harder than hitting Merge, it creates one commit, and it'll at least use the PR title and list of commit details instead of an uninformative message like "Merged pull request XXX", which _really_ frustrated me that other day when I was trying to find out what merges could've impacted our logging setup...

I think that's still everything you love about merge commits. What actually makes merge commits unique (preserving and _honest_ history) isn't what I think you're arguing for.

Reply:
Yeah I think I focused on rebase too much here. I think squash would be better than both merge commits and rebase now that you pointed it out. :+1: (I'm not taking a hard position here. If the team prefers any one of these I'm fine with it.)

Reply:
Yeah, sounds like you are more arguing against "bad" commits than against rebase.

Reply:
> With a rebase merge that becomes harder and requires more digging.

You probably know this, and still consider it "more digging" but if you open the commit on GitHub (clicking through from Circle or Jenkins), GH has a link to the PR in the header.

Reply:
My problem with squash is that it ends up diluting a clean commit stream. I attempt to make each of my commits atomic and include a description of what/why everything was done. When you squash it dilutes the change set. Now a specific commit message will be interspersed with many others and it will be hard to tell which what/why is applicable.

Reply:
Let's be careful here. This conversation tends to drift towards either-or, but we can all be happy very easily:

- If you like to rebase to a detailed multi-commit PR with good content, please do so
- If you don't, please just use Squash instead of Merge

Reply:
Oh totally agree, I'm just shaving yaks at this point.